core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,190 @@
+        return Literal.of(defaultValue.textValue()).to(type).value();
+      case FIXED:
+        Preconditions.checkArgument(
+            defaultValue.isTextual() && defaultValue.textValue().length() == ((Types.FixedType) type).length() * 2,
+            "Cannot parse %s to a %s value",
+            defaultValue,
+            type);
+        byte[] fixedBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.allocate(((Types.FixedType) type).length()).put(fixedBytes);
+      case BINARY:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        byte[] binaryBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.wrap(binaryBytes);
+      case LIST:
+        Preconditions.checkArgument(defaultValue.isArray(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        List<Object> defaultList = Lists.newArrayList();
+        for (JsonNode element : defaultValue) {
+          defaultList.add(parseDefaultFromJson(type.asListType().elementType(), element));
+        }
+        return defaultList;
+      case MAP:
+        Preconditions.checkArgument(
+            defaultValue.isObject() && defaultValue.has("keys") && defaultValue.has("values") &&
+                defaultValue.get("keys").isArray() && defaultValue.get("values").isArray(),
+            "Cannot parse %s to a %s value",
+            defaultValue,
+            type);
+        Map<Object, Object> defaultMap = Maps.newHashMap();
+        JsonNode keys = defaultValue.get("keys");
+        JsonNode values = defaultValue.get("values");
+        List<JsonNode> keyList = Lists.newArrayList(keys.iterator());
+        List<JsonNode> valueList = Lists.newArrayList(values.iterator());
+
+        for (int i = 0; i < keyList.size(); i++) {
+          defaultMap.put(
+              parseDefaultFromJson(type.asMapType().keyType(), keyList.get(i)),
+              parseDefaultFromJson(type.asMapType().valueType(), valueList.get(i)));
+        }
+        return defaultMap;
+      case STRUCT:
+        Preconditions.checkArgument(defaultValue.isObject(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        Map<Integer, Object> defaultStruct = Maps.newHashMap();
+        for (Types.NestedField subField : type.asStructType().fields()) {
+          String fieldIdAsString = String.valueOf(subField.fieldId());
+          Object value = defaultValue.has(fieldIdAsString) ? parseDefaultFromJson(
+              subField.type(),
+              defaultValue.get(fieldIdAsString)) : null;
+          if (value != null) {
+            defaultStruct.put(subField.fieldId(), value);
+          }
+        }
+        return defaultStruct;
+      default:
+        return null;
+    }
+  }
+
+  public static Object convertJavaDefaultForSerialization(Type type, Object value) {
```

Comment:
Like the other parsers, this method should be passed a `JsonGenerator` that handles creating the JSON string.

Reply:
I refactored it to be:

Reply:
@rzhang10, please look at the other parsers and match what they do. You should be using a `JsonGenerator`.